airflow/www/static/js/trigger.js

```diff
@@ -20,13 +20,24 @@
 /* global document, CodeMirror, window */
 
 const textArea = document.getElementById('json');
+const recentConfigList = document.getElementById('recent_configs');
 const minHeight = 300;
 const maxHeight = window.innerHeight - 450;
 const height = maxHeight > minHeight ? maxHeight : minHeight;
 
 CodeMirror.fromTextArea(textArea, {
   lineNumbers: true,
-  mode: { name: 'javascript', json: true },
+  mode: {
+    name: 'javascript',
+    json: true,
+  },
   gutters: ['CodeMirror-lint-markers'],
   lint: true,
-}).setSize(null, height);
+})
+  .setSize(null, height);
+
+function setRecentConfig(e) {
+  document.querySelector('.CodeMirror').CodeMirror.setValue(e.target.value);
```

Comment:
I wonder if we should try to format the config JSON so it doesn't appear as a single line

```
function setRecentConfig(e) {
  let { value } = e.target;
  try {
    const json = JSON.parse(value);
    value = JSON.stringify(json, null, 2);
  } catch (err) {
    console.err(err);
  }
  document.querySelector('.CodeMirror').CodeMirror.setValue(value);
}
```

Reply:
Hey yeah! I wanted to do this as well, but code formatting is a function that CodeMirror no longer supports. The other alternative is adding another JS library just to do this formatting, but I don't think we need this overhead (let me know if we already have one that can do it!).

The other alternative is writing a bespoke function to do JSON formatting but I didn't think it was worth the trouble.

Reply:
The function above should do it. We just need that try/catch block to capture thrown errors when it is not valid json.